.github/workflows/changelog-checker.yml

```diff
@@ -0,0 +1,48 @@
+# This workflow checks that there is either a 'pr/no-changelog' label applied to a PR
+# or there is a changelog/<pr number>.txt file associated with a PR for a changelog entry
+
+name: Check Changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to master and all release branches
+    branches:
+      - master/*
+      - release/*
+
+jobs:
+  # checks that a changelog entry is present for a PR
+  changelog-check:
+    # If there  a `pr/no-changelog` label we ignore this check
+    if: "!contains(github.event.pull_request.labels.*.name, 'pr/no-changelog')"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Check for changelog entry in diff
+        run: |
+          # check if there is a diff in the changelog directory
+          changelog_files=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/${{ github.event.pull_request.base.ref }}")" -- changelog/${{ github.event.pull_request.number }}.txt)
+
+          # If we do not find a file in changelog/, we fail the check
+          if [ -z "$changelog_files" ]; then
+            # Fail status check when no changelog entry was found on the PR
+            echo "Did not find a changelog entry and the 'pr/no-changelog' label was not applied. Reference - https://github.com/hashicorp/vault/pull/10363"
+            exit 1
+          else
+            # Don't want square brackets bc not using test
+            if grep -q ':enhancement$' $changelog_files; then
```

Comment:
Another common one that I've seen in the past is `bug` (correct) vs `bugs` (incorrect).